activerecord/lib/active_record/secure_token.rb

```diff
@@ -26,8 +26,11 @@ module ClassMethods
       def has_secure_token(attribute = :token)
         # Load securerandom only when has_secure_key is used.
         require 'active_support/core_ext/securerandom'
-        define_method("regenerate_#{attribute}") { update! attribute => self.class.generate_unique_secure_token }
-        before_create { self.send("#{attribute}=", self.class.generate_unique_secure_token) }
+        unless self.method_defined?("generate_#{attribute}")
```

Comment:
I don't understand why we're checking for this first? Let's just let Let's Definition Win. If you declare has_secure_token at the top of the file, then you can just set generate_token yourself later, and it'll overwrite, and all will work.